python/python/lance/torch/kmeans.py

```diff
@@ -205,6 +211,7 @@ def _fit_once(
 
         new_centroids = torch.zeros_like(self.centroids, device=self.device)
         counts_per_part = torch.zeros(self.centroids.shape[0], device=self.device)
+        ones = torch.ones(1024 * 16, device=self.device)
```

Comment:
where does this and the 10240 constant from before come from?

Reply:
This is just an estmatiation from the normal batch size we would like to have.
So if the batch size is smaller than 1024 * 16 (which takes 64KB cuda / mps memory) , the allocation only happens once during one training loop. 

If the 1024 * 16 is smaller than the batch size, later the loop will adjust the allocation as well.